src/Analyser/NodeScopeResolver.php

```diff
@@ -4177,6 +4190,49 @@ public function getPhpDocs(Scope $scope, Node\FunctionLike|Node\Stmt\Property $n
 		return [$templateTypeMap, $phpDocParameterTypes, $phpDocReturnType, $phpDocThrowType, $deprecatedDescription, $isDeprecated, $isInternal, $isFinal, $isPure, $acceptsNamedArguments, $isReadOnly, $docComment, $asserts, $selfOutType];
 	}
 
+	/**
+	 * @return array<string, Type>
+	 */
+	private function getParameterOutTypes(FunctionReflection|MethodReflection $reflection, Scope $scope): array
+	{
+		$class = null;
+		$trait = null;
+		if ($reflection instanceof MethodReflection) {
+			$class = $reflection->getDeclaringClass()->getName();
+			$file = $reflection->getDeclaringClass()->getFileName();
+		} else {
+			$file = $reflection->getFileName();
+		}
+
+		$resolvedPhpDoc = null;
+		$functionName = $reflection->getName();
+		$docComment = $reflection->getDocComment();
+
+		if ($docComment !== null) {
+			$resolvedPhpDoc = $this->fileTypeMapper->getResolvedPhpDoc(
```

Comment:
`FunctionReflection|ExtendedMethodReflection` should already have a method for `@param-out` tags. But to be honest, it should probably be an information for a specific parameter on `ParameterReflectionWithPhpDocs` to be really clean. Something like `public function getOutType(): ?Type`.

Reply:
You shouldn't definitely parse PHPDoc here again. That's a job for PhpClassReflectionExtension, `BetterReflectionProvider::getCustomFunction()` and `NativeFunctionReflectionProvider`.